## Analysis of how the Derby/Rivalry Factor affects Football Match Outcomes

### Data from English & Scottish Football League & Top Tier Spain, Germany & Italy, 2003-2022
#### Rivalries sourced from:
#### UK: https://en.wikipedia.org/wiki/List_of_association_football_rivalries_in_the_United_Kingdom
#### Germany: https://en.wikipedia.org/wiki/German_football_rivalries
#### Italy: https://en.wikipedia.org/wiki/Football_derbies_in_Italy
#### Spain: https://en.wikipedia.org/wiki/Spanish_football_rivalries
##### In total the sample size for all matches is 76,184 of which 3,355 have been identified as a derby of above zero magnitude
##### The terms 'rivalry' and 'derby' are used interchangeably and deemed equal for the purposes of this analysis

In [1]:
#import libraries
import pandas as pd
import numpy as np

pd.set_option('display.max_columns', None)
pd.options.mode.chained_assignment = None  

#### Read in pre-processed CSV of results and match stats of all major European leagues 2003-2022

In [2]:
df = pd.read_csv('C:\\Users\\Betting\\Football-Data-Combined\\Leagues_Data_Combined.csv', index_col=[0])


In [3]:
df.head()

Div        Date HomeTeam     AwayTeam  FTHG  FTAG  HTHG  HTAG  HST  AST  \
0  I1  21/08/2021    Inter        Genoa   4.0   0.0   2.0   0.0  8.0  5.0   
1  I1  21/08/2021   Verona     Sassuolo   2.0   3.0   0.0   1.0  4.0  8.0   
2  I1  21/08/2021   Empoli        Lazio   1.0   3.0   1.0   3.0  5.0  5.0   
3  I1  21/08/2021   Torino     Atalanta   1.0   2.0   0.0   1.0  8.0  2.0   
4  I1  22/08/2021  Bologna  Salernitana   3.0   2.0   0.0   0.0  7.0  4.0   

     HF    AF   HC   AC   HY   AY   HR   AR   TG   TST    TF    TC   TY   TR  \
0  18.0   7.0  8.0  2.0  1.0  2.0  0.0  0.0  4.0  13.0  25.0  10.0  3.0  0.0   
1  11.0  12.0  7.0  4.0  3.0  2.0  1.0  0.0  5.0  12.0  23.0  11.0  5.0  1.0   
2  13.0  10.0  5.0  3.0  2.0  1.0  0.0  0.0  4.0  10.0  23.0   8.0  3.0  0.0   
3  17.0  13.0  5.0  1.0  2.0  2.0  0.0  0.0  3.0  10.0  30.0   6.0  4.0  0.0   
4  13.0  15.0  9.0  4.0  6.0  3.0  2.0  1.0  5.0  11.0  28.0  13.0  9.0  3.0   

   HW_%   X_%  AW_%  Ov2.5_%  Un2.5_%   Exp  HG_Exp  AG_Exp  Sup Exp  
0  0.71  0.19  0.11     0.58     0.42  3.03    2.44    0.62     1.82  
1  0.37  0.28  0.36     0.52     0.48  2.76    1.41    1.38     0.03  
2  0.20  0.24  0.55     0.56     0.44  2.94    0.94    1.97    -1.03  
3  0.18  0.23  0.60     0.61     0.39  3.14    0.93    2.25    -1.32  
4  0.58  0.24  0.19     0.53     0.47  2.79    1.95    0.86     1.09

#### Observe general descriptive stats

In [4]:
df.describe()

FTHG           FTAG           HTHG           HTAG  \
count  141422.000000  141422.000000  141353.000000  141352.000000   
mean        1.476284       1.134809       0.651398       0.494956   
std         1.246980       1.107168       0.806706       0.708830   
min         0.000000       0.000000       0.000000       0.000000   
25%         1.000000       0.000000       0.000000       0.000000   
50%         1.000000       1.000000       0.000000       0.000000   
75%         2.000000       2.000000       1.000000       1.000000   
max        10.000000      13.000000       7.000000       7.000000   

                HST           AST            HF            AF            HC  \
count  85907.000000  85910.000000  85206.000000  85206.000000  86453.000000   
mean       5.178018      4.197986     12.847030     13.317536      5.696517   
std        2.803821      2.466060      4.544789      4.640416      2.933241   
min        0.000000      0.000000      0.000000      0.000000      0.000000   
25%        3.000000      2.000000     10.000000     10.000000      4.000000   
50%        5.000000      4.000000     12.000000     13.000000      5.000000   
75%        7.000000      6.000000     16.000000     16.000000      7.000000   
max       27.000000     23.000000     48.000000     77.000000     26.000000   

                 AC            HY            AY            HR            AR  \
count  86453.000000  86846.000000  86847.000000  86847.000000  86846.000000   
mean       4.636265      1.668528      1.975578      0.087395      0.119867   
std        2.618209      1.299118      1.367905      0.299665      0.351243   
min        0.000000      0.000000      0.000000      0.000000      0.000000   
25%        3.000000      1.000000      1.000000      0.000000      0.000000   
50%        4.000000      2.000000      2.000000      0.000000      0.000000   
75%        6.000000      2.000000      3.000000      0.000000      0.000000   
max       21.000000     11.000000      9.000000      3.000000      9.000000   

                  TG           TST            TF            TC            TY  \
count  141422.000000  85907.000000  85206.000000  86453.000000  86846.000000   
mean        2.611093      9.375895     26.164566     10.332782      3.644071   
std         1.641780      3.750029      7.522284      3.535226      2.100462   
min         0.000000      0.000000      0.000000      0.000000      0.000000   
25%         1.000000      7.000000     21.000000      8.000000      2.000000   
50%         2.000000      9.000000     25.000000     10.000000      3.000000   
75%         4.000000     12.000000     31.000000     13.000000      5.000000   
max        13.000000     33.000000     90.000000     30.000000     15.000000   

                 TR           HW_%            X_%           AW_%  \
count  86846.000000  122291.000000  122291.000000  122291.000000   
mean       0.207263       0.437061       0.269301       0.293615   
std        0.478151       0.142725       0.039868       0.128539   
min        0.000000       0.030000       0.050000       0.020000   
25%        0.000000       0.350000       0.260000       0.210000   
50%        0.000000       0.430000       0.280000       0.280000   
75%        0.000000       0.520000       0.290000       0.360000   
max       10.000000       0.930000       0.770000       0.900000   

             Ov2.5_%        Un2.5_%            Exp         HG_Exp  \
count  122241.000000  122241.000000  121848.000000  121848.000000   
mean        0.488177       0.511826       2.633684       1.508862   
std         0.071264       0.071264       0.286418       0.438592   
min         0.270000       0.140000       1.820000       0.220000   
25%         0.440000       0.470000       2.440000       1.240000   
50%         0.480000       0.520000       2.600000       1.440000   
75%         0.530000       0.560000       2.790000       1.710000   
max         0.860000       0.730000       3.820000       3.570000   

              

#### Extract leagues to analyse. Derbies have been logged for UK football league and top division Spain, Italy & Germany so isolate data for those divisions only.

In [5]:
df_prem = df[(df['Div'] == 'E1')]
df_champ = df[(df['Div'] == 'E2')]
df_l1 = df[(df['Div'] == 'E3')]
df_l2 = df[(df['Div'] == 'E4')]
df_It1 = df[(df['Div'] == 'I1')]
df_G1 = df[(df['Div'] == 'DF1')]
df_Sp1 = df[(df['Div'] == 'SP1')]
df_Sc1 = df[(df['Div'] == 'SC0')]
df_Sc2 = df[(df['Div'] == 'SC1')]
df_Sc3 = df[(df['Div'] == 'SC2')]



#### Merge these separated divisions into a single DF, df1

In [6]:
frames = [df_prem, df_champ,df_l1, df_l2, df_l2, df_It1, df_G1, df_Sp1, df_Sc1, df_Sc2, df_Sc3 ]
df1 = pd.concat(frames)
df1.reset_index(inplace=True)
df1.head()


index Div        Date    HomeTeam        AwayTeam  FTHG  FTAG  HTHG  HTAG  \
0   7796  E1  13/08/2021   Brentford         Arsenal   2.0   0.0   1.0   0.0   
1   7797  E1  14/08/2021  Man United           Leeds   5.0   1.0   1.0   0.0   
2   7798  E1  14/08/2021     Burnley        Brighton   1.0   2.0   1.0   0.0   
3   7799  E1  14/08/2021     Chelsea  Crystal Palace   3.0   0.0   2.0   0.0   
4   7800  E1  14/08/2021     Everton     Southampton   3.0   1.0   0.0   1.0   

   HST  AST    HF    AF   HC   AC   HY   AY   HR   AR   TG   TST    TF    TC  \
0  3.0  4.0  12.0   8.0  2.0  5.0  0.0  0.0  0.0  0.0  2.0   7.0  20.0   7.0   
1  8.0  3.0  11.0   9.0  5.0  4.0  1.0  2.0  0.0  0.0  6.0  11.0  20.0   9.0   
2  3.0  8.0  10.0   7.0  7.0  6.0  2.0  1.0  0.0  0.0  3.0  11.0  17.0  13.0   
3  6.0  1.0  15.0  11.0  5.0  2.0  0.0  0.0  0.0  0.0  3.0   7.0  26.0   7.0   
4  6.0  3.0  13.0  15.0  6.0  8.0  2.0  0.0  0.0  0.0  4.0   9.0  28.0  14.0   

    TY   TR  HW_%   X_%  AW_%  Ov2.5_%  Un2.5_%   Exp  HG_Exp  AG_Exp  Sup Exp  
0  0.0  0.0  0.24  0.28  0.48     0.44     0.56  2.44    0.93    1.51    -0.58  
1  3.0  0.0  0.62  0.21  0.16     0.58     0.42  3.03    2.20    0.80     1.40  
2  3.0  0.0  0.30  0.31  0.39     0.39     0.61  2.25    1.02    1.23    -0.21  
3  0.0  0.0  0.76  0.16  0.08     0.54     0.46  2.84    2.39    0.45     1.94  
4  2.0  0.0  0.49  0.27  0.24     0.46     0.54  2.53    1.58    0.95     0.63

#### Read in csv containing pre-processed list of European derbies

In [7]:
df_derb = pd.read_csv('C:\\Users\\Betting\\DERBIES.csv')
df_derb.head(10)

H_Team       A_Team
0              Oxford      Swindon
1  Milton Keynes Dons      Wycombe
2           Cambridge    Peterboro
3             Reading      Swindon
4              Oxford      Reading
5               Luton    Stevenage
6               Luton      Watford
7  Milton Keynes Dons  Northampton
8  Milton Keynes Dons    Peterboro
9         Northampton    Peterboro

In [8]:
df_derb.count

<bound method DataFrame.count of                  H_Team      A_Team
0                Oxford     Swindon
1    Milton Keynes Dons     Wycombe
2             Cambridge   Peterboro
3               Reading     Swindon
4                Oxford     Reading
..                  ...         ...
240        Queen of Sth   Stranraer
241        Queen of Sth         Ayr
242        Queen of Sth  Kilmarnock
243           Stranraer         Ayr
244                 Ayr  Kilmarnock

[245 rows x 2 columns]>

##### ** 245 fixtures highlighted as having some level of rivalry **

#### Create a new column in main DF named'Derby'. Add default value as 0 (assume each match has zero rivalry initially)
#### Loop through the list of derbies DF. When a derby fixture in the main DF is identified, add value of 1 to 'Derby' column

In [9]:
# Initialize a new column 'Derby' in df_eng with default value 0
df1['Derby'] = 0

# Iterate over each row in df_eng
for _, row in df1.iterrows():
    home_team = row['HomeTeam']
    away_team = row['AwayTeam']
    
    # Check if both home_team and away_team exist in df_derb['Team_A'] and df_derb['Team_B']
    if (((df_derb['H_Team'] == home_team) & (df_derb['A_Team'] == away_team)).any()) or \
       (((df_derb['H_Team'] == away_team) & (df_derb['A_Team'] == home_team)).any()):
        df1.loc[_, 'Derby'] = 1


In [10]:
df1

index  Div        Date    HomeTeam        AwayTeam  FTHG  FTAG  HTHG  \
0       7796   E1  13/08/2021   Brentford         Arsenal   2.0   0.0   1.0   
1       7797   E1  14/08/2021  Man United           Leeds   5.0   1.0   1.0   
2       7798   E1  14/08/2021     Burnley        Brighton   1.0   2.0   1.0   
3       7799   E1  14/08/2021     Chelsea  Crystal Palace   3.0   0.0   2.0   
4       7800   E1  14/08/2021     Everton     Southampton   3.0   1.0   0.0   
...      ...  ...         ...         ...             ...   ...   ...   ...   
76179  60198  SC2  29/04/2006   Dumbarton         Partick   2.0   3.0   2.0   
76180  60199  SC2  29/04/2006      Forfar          Morton   0.0   2.0   0.0   
76181  60200  SC2  29/04/2006   Peterhead             Ayr   1.0   2.0   0.0   
76182  60201  SC2  29/04/2006   Raith Rvs           Alloa   0.0   1.0   0.0   
76183  60202  SC2  29/04/2006    Stirling          Gretna   2.0   1.0   2.0   

       HTAG  HST  AST    HF    AF   HC   AC   HY   AY   HR   AR   TG   TST  \
0       0.0  3.0  4.0  12.0   8.0  2.0  5.0  0.0  0.0  0.0  0.0  2.0   7.0   
1       0.0  8.0  3.0  11.0   9.0  5.0  4.0  1.0  2.0  0.0  0.0  6.0  11.0   
2       0.0  3.0  8.0  10.0   7.0  7.0  6.0  2.0  1.0  0.0  0.0  3.0  11.0   
3       0.0  6.0  1.0  15.0  11.0  5.0  2.0  0.0  0.0  0.0  0.0  3.0   7.0   
4       1.0  6.0  3.0  13.0  15.0  6.0  8.0  2.0  0.0  0.0  0.0  4.0   9.0   
...     ...  ...  ...   ...   ...  ...  ...  ...  ...  ...  ...  ...   ...   
76179   0.0  NaN  NaN   NaN   NaN  NaN  NaN  NaN  NaN  NaN  NaN  5.0   NaN   
76180   0.0  NaN  NaN   NaN   NaN  NaN  NaN  NaN  NaN  NaN  NaN  2.0   NaN   
76181   0.0  NaN  NaN   NaN   NaN  NaN  NaN  NaN  NaN  NaN  NaN  3.0   NaN   
76182   0.0  NaN  NaN   NaN   NaN  NaN  NaN  NaN  NaN  NaN  NaN  1.0   NaN   
76183   0.0  NaN  NaN   NaN   NaN  NaN  NaN  NaN  NaN  NaN  NaN  3.0   NaN   

         TF    TC   TY   TR  HW_%   X_%  AW_%  Ov2.5_%  Un2.5_%   Exp  HG_Exp  \
0      20.0   7.0  0.0  0.0  0.24  0.28  0.48     0.44     0.56  2.44    0.93   
1      20.0   9.0  3.0  0.0  0.62  0.21  0.16     0.58     0.42  3.03    2.20   
2      17.0  13.0  3.0  0.0  0.30  0.31  0.39     0.39     0.61  2.25    1.02   
3      26.0   7.0  0.0  0.0  0.76  0.16  0.08     0.54     0.46  2.84    2.39   
4      28.0  14.0  2.0  0.0  0.49  0.27  0.24     0.46     0.54  2.53    1.58   
...     ...   ...  ...  ...   ...   ...   ...      ...      ...   ...     ...   
76179   NaN   NaN  NaN  NaN  0.40  0.26  0.34     0.52     0.48  2.76    1.46   
76180   NaN   NaN  NaN  NaN  0.41  0.27  0.33     0.52     0.48  2.76    1.50   
76181   NaN   NaN  NaN  NaN  0.52  0.25  0.22     0.56     0.44  2.94    1.90   
76182   NaN   NaN  NaN  NaN  0.35  0.28  0.37     0.52     0.48  2.76    1.35   
76183   NaN   NaN  NaN  NaN  0.17  0.23  0.60     0.58     0.42  3.03    0.86   

       AG_Exp  Sup Exp  Derby  
0        1.51    -0.58      0  
1        0.80     1.40      1  
2        1.23    -0.21      0  
3        0.45     1.94      0  
4        0.95     0.63      0  
...       ...      ...    ...  
76179    1.30     0.16      0  
76180    1.28     0.22      0  
76181    1.01     0.89      0  
76182    1.41    -0.06      0  
76183    2.17    -1.31      0  

[76184 rows x 35 columns]

#### Add a new column providing each match outcome. Firstly define a function, then create a new column and apply the function

In [11]:
# create new column in df1
def f(row):
    if row['FTHG'] == row['FTAG']:
        val = 0
    elif row['FTHG'] > row['FTAG']:
        val = 1
    else:
        val = 2
    return val
        

In [12]:
df1['Res'] = df1.apply(f, axis=1)
df1

index  Div        Date    HomeTeam        AwayTeam  FTHG  FTAG  HTHG  \
0       7796   E1  13/08/2021   Brentford         Arsenal   2.0   0.0   1.0   
1       7797   E1  14/08/2021  Man United           Leeds   5.0   1.0   1.0   
2       7798   E1  14/08/2021     Burnley        Brighton   1.0   2.0   1.0   
3       7799   E1  14/08/2021     Chelsea  Crystal Palace   3.0   0.0   2.0   
4       7800   E1  14/08/2021     Everton     Southampton   3.0   1.0   0.0   
...      ...  ...         ...         ...             ...   ...   ...   ...   
76179  60198  SC2  29/04/2006   Dumbarton         Partick   2.0   3.0   2.0   
76180  60199  SC2  29/04/2006      Forfar          Morton   0.0   2.0   0.0   
76181  60200  SC2  29/04/2006   Peterhead             Ayr   1.0   2.0   0.0   
76182  60201  SC2  29/04/2006   Raith Rvs           Alloa   0.0   1.0   0.0   
76183  60202  SC2  29/04/2006    Stirling          Gretna   2.0   1.0   2.0   

       HTAG  HST  AST    HF    AF   HC   AC   HY   AY   HR   AR   TG   TST  \
0       0.0  3.0  4.0  12.0   8.0  2.0  5.0  0.0  0.0  0.0  0.0  2.0   7.0   
1       0.0  8.0  3.0  11.0   9.0  5.0  4.0  1.0  2.0  0.0  0.0  6.0  11.0   
2       0.0  3.0  8.0  10.0   7.0  7.0  6.0  2.0  1.0  0.0  0.0  3.0  11.0   
3       0.0  6.0  1.0  15.0  11.0  5.0  2.0  0.0  0.0  0.0  0.0  3.0   7.0   
4       1.0  6.0  3.0  13.0  15.0  6.0  8.0  2.0  0.0  0.0  0.0  4.0   9.0   
...     ...  ...  ...   ...   ...  ...  ...  ...  ...  ...  ...  ...   ...   
76179   0.0  NaN  NaN   NaN   NaN  NaN  NaN  NaN  NaN  NaN  NaN  5.0   NaN   
76180   0.0  NaN  NaN   NaN   NaN  NaN  NaN  NaN  NaN  NaN  NaN  2.0   NaN   
76181   0.0  NaN  NaN   NaN   NaN  NaN  NaN  NaN  NaN  NaN  NaN  3.0   NaN   
76182   0.0  NaN  NaN   NaN   NaN  NaN  NaN  NaN  NaN  NaN  NaN  1.0   NaN   
76183   0.0  NaN  NaN   NaN   NaN  NaN  NaN  NaN  NaN  NaN  NaN  3.0   NaN   

         TF    TC   TY   TR  HW_%   X_%  AW_%  Ov2.5_%  Un2.5_%   Exp  HG_Exp  \
0      20.0   7.0  0.0  0.0  0.24  0.28  0.48     0.44     0.56  2.44    0.93   
1      20.0   9.0  3.0  0.0  0.62  0.21  0.16     0.58     0.42  3.03    2.20   
2      17.0  13.0  3.0  0.0  0.30  0.31  0.39     0.39     0.61  2.25    1.02   
3      26.0   7.0  0.0  0.0  0.76  0.16  0.08     0.54     0.46  2.84    2.39   
4      28.0  14.0  2.0  0.0  0.49  0.27  0.24     0.46     0.54  2.53    1.58   
...     ...   ...  ...  ...   ...   ...   ...      ...      ...   ...     ...   
76179   NaN   NaN  NaN  NaN  0.40  0.26  0.34     0.52     0.48  2.76    1.46   
76180   NaN   NaN  NaN  NaN  0.41  0.27  0.33     0.52     0.48  2.76    1.50   
76181   NaN   NaN  NaN  NaN  0.52  0.25  0.22     0.56     0.44  2.94    1.90   
76182   NaN   NaN  NaN  NaN  0.35  0.28  0.37     0.52     0.48  2.76    1.35   
76183   NaN   NaN  NaN  NaN  0.17  0.23  0.60     0.58     0.42  3.03    0.86   

       AG_Exp  Sup Exp  Derby  Res  
0        1.51    -0.58      0    1  
1        0.80     1.40      1    1  
2        1.23    -0.21      0    2  
3        0.45     1.94      0    1  
4        0.95     0.63      0    1  
...       ...      ...    ...  ...  
76179    1.30     0.16      0    2  
76180    1.28     0.22      0    2  
76181    1.01     0.89      0    2  
76182    1.41    -0.06      0    2  
76183    2.17    -1.31      0    1  

[76184 rows x 36 columns]

#### Create a DF of derby matches only

In [13]:
derbies = df1[df1['Derby'] == 1]
derbies.reset_index(inplace=True)
derbies

level_0  index  Div        Date    HomeTeam   AwayTeam  FTHG  FTAG  \
0           1   7797   E1  14/08/2021  Man United      Leeds   5.0   1.0   
1          18   7814   E1  22/08/2021     Arsenal    Chelsea   0.0   2.0   
2          26   7822   E1  28/08/2021   Liverpool    Chelsea   1.0   1.0   
3          49   7845   E1  19/09/2021   Tottenham    Chelsea   0.0   3.0   
4          58   7854   E1  26/09/2021     Arsenal  Tottenham   3.0   1.0   
...       ...    ...  ...         ...         ...        ...   ...   ...   
3350    75960  59979  SC2  03/03/2007   Stranraer        Ayr   0.0   3.0   
3351    76023  60042  SC2  27/08/2005    Stirling      Alloa   1.0   2.0   
3352    76044  60063  SC2  07/10/2005    Stirling      Alloa   0.0   0.0   
3353    76065  60084  SC2  05/11/2005       Alloa   Stirling   2.0   4.0   
3354    76154  60173  SC2  28/03/2006       Alloa   Stirling   0.0   0.0   

      HTHG  HTAG  HST   AST    HF    AF    HC    AC   HY   AY   HR   AR   TG  \
0      1.0   0.0  8.0   3.0  11.0   9.0   5.0   4.0  1.0  2.0  0.0  0.0  6.0   
1      0.0   2.0  3.0   5.0  10.0   4.0   9.0   8.0  3.0  0.0  0.0  0.0  2.0   
2      1.0   1.0  7.0   3.0  13.0   4.0  12.0   3.0  0.0  2.0  0.0  1.0  2.0   
3      0.0   0.0  2.0  10.0   4.0  15.0   5.0  11.0  0.0  1.0  0.0  0.0  3.0   
4      3.0   0.0  7.0   4.0  12.0  13.0   4.0   4.0  2.0  1.0  0.0  0.0  4.0   
...    ...   ...  ...   ...   ...   ...   ...   ...  ...  ...  ...  ...  ...   
3350   0.0   3.0  NaN   NaN   NaN   NaN   NaN   NaN  NaN  NaN  NaN  NaN  3.0   
3351   0.0   1.0  NaN   NaN   NaN   NaN   NaN   NaN  NaN  NaN  NaN  NaN  3.0   
3352   0.0   0.0  NaN   NaN   NaN   NaN   NaN   NaN  NaN  NaN  NaN  NaN  0.0   
3353   0.0   3.0  NaN   NaN   NaN   NaN   NaN   NaN  NaN  NaN  NaN  NaN  6.0   
3354   0.0   0.0  NaN   NaN   NaN   NaN   NaN   NaN  NaN  NaN  NaN  NaN  0.0   

       TST    TF    TC   TY   TR  HW_%   X_%  AW_%  Ov2.5_%  Un2.5_%   Exp  \
0     11.0  20.0   9.0  3.0  0.0  0.62  0.21  0.16     0.58     0.42  3.03   
1      8.0  14.0  17.0  3.0  0.0  0.20  0.26  0.54     0.46     0.54  2.53   
2     10.0  17.0  15.0  2.0  1.0  0.38  0.29  0.33     0.48     0.52  2.60   
3     12.0  19.0  16.0  1.0  0.0  0.20  0.25  0.55     0.46     0.54  2.53   
4     11.0  25.0   8.0  3.0  0.0  0.42  0.29  0.29     0.46     0.54  2.53   
...    ...   ...   ...  ...  ...   ...   ...   ...      ...      ...   ...   
3350   NaN   NaN   NaN  NaN  NaN  0.41  0.26  0.33     0.48     0.52  2.60   
3351   NaN   NaN   NaN  NaN  NaN  0.40  0.26  0.34     0.50     0.50  2.66   
3352   NaN   NaN   NaN  NaN  NaN  0.37  0.26  0.37     0.46     0.54  2.53   
3353   NaN   NaN   NaN  NaN  NaN  0.37  0.26  0.37     0.53     0.47  2.79   
3354   NaN   NaN   NaN  NaN  NaN  0.27  0.26  0.47     0.55     0.45  2.89   

      HG_Exp  AG_Exp  Sup Exp  Derby  Res  
0       2.20    0.80     1.40      1    1  
1       0.83    1.70    -0.87      1    2  
2       1.37    1.23     0.14      1    0  
3       0.82    1.71    -0.89      1    2  
4       1.43    1.10     0.33      1    1  
...      ...     ...      ...    ...  ...  
3350    1.40    1.20     0.20      1    2  
3351    1.41    1.25     0.16      1    2  
3352    1.26    1.26     0.00      1    0  
3353    1.40    1.40     0.00      1    2  
3354    1.16    1.73    -0.57      1    0  

[3355 rows x 37 columns]

#### Create a DF of non-derby matches only

In [14]:
non_derbies = df1[df1['Derby'] == 0]
non_derbies.reset_index(inplace=True)
non_derbies

level_0  index  Div        Date   HomeTeam        AwayTeam  FTHG  FTAG  \
0            0   7796   E1  13/08/2021  Brentford         Arsenal   2.0   0.0   
1            2   7798   E1  14/08/2021    Burnley        Brighton   1.0   2.0   
2            3   7799   E1  14/08/2021    Chelsea  Crystal Palace   3.0   0.0   
3            4   7800   E1  14/08/2021    Everton     Southampton   3.0   1.0   
4            5   7801   E1  14/08/2021  Leicester          Wolves   1.0   0.0   
...        ...    ...  ...         ...        ...             ...   ...   ...   
72824    76179  60198  SC2  29/04/2006  Dumbarton         Partick   2.0   3.0   
72825    76180  60199  SC2  29/04/2006     Forfar          Morton   0.0   2.0   
72826    76181  60200  SC2  29/04/2006  Peterhead             Ayr   1.0   2.0   
72827    76182  60201  SC2  29/04/2006  Raith Rvs           Alloa   0.0   1.0   
72828    76183  60202  SC2  29/04/2006   Stirling          Gretna   2.0   1.0   

       HTHG  HTAG  HST  AST    HF    AF   HC   AC   HY   AY   HR   AR   TG  \
0       1.0   0.0  3.0  4.0  12.0   8.0  2.0  5.0  0.0  0.0  0.0  0.0  2.0   
1       1.0   0.0  3.0  8.0  10.0   7.0  7.0  6.0  2.0  1.0  0.0  0.0  3.0   
2       2.0   0.0  6.0  1.0  15.0  11.0  5.0  2.0  0.0  0.0  0.0  0.0  3.0   
3       0.0   1.0  6.0  3.0  13.0  15.0  6.0  8.0  2.0  0.0  0.0  0.0  4.0   
4       1.0   0.0  5.0  3.0   6.0  10.0  5.0  4.0  1.0  2.0  0.0  0.0  1.0   
...     ...   ...  ...  ...   ...   ...  ...  ...  ...  ...  ...  ...  ...   
72824   2.0   0.0  NaN  NaN   NaN   NaN  NaN  NaN  NaN  NaN  NaN  NaN  5.0   
72825   0.0   0.0  NaN  NaN   NaN   NaN  NaN  NaN  NaN  NaN  NaN  NaN  2.0   
72826   0.0   0.0  NaN  NaN   NaN   NaN  NaN  NaN  NaN  NaN  NaN  NaN  3.0   
72827   0.0   0.0  NaN  NaN   NaN   NaN  NaN  NaN  NaN  NaN  NaN  NaN  1.0   
72828   2.0   0.0  NaN  NaN   NaN   NaN  NaN  NaN  NaN  NaN  NaN  NaN  3.0   

        TST    TF    TC   TY   TR  HW_%   X_%  AW_%  Ov2.5_%  Un2.5_%   Exp  \
0       7.0  20.0   7.0  0.0  0.0  0.24  0.28  0.48     0.44     0.56  2.44   
1      11.0  17.0  13.0  3.0  0.0  0.30  0.31  0.39     0.39     0.61  2.25   
2       7.0  26.0   7.0  0.0  0.0  0.76  0.16  0.08     0.54     0.46  2.84   
3       9.0  28.0  14.0  2.0  0.0  0.49  0.27  0.24     0.46     0.54  2.53   
4       8.0  16.0   9.0  3.0  0.0  0.56  0.26  0.18     0.47     0.53  2.56   
...     ...   ...   ...  ...  ...   ...   ...   ...      ...      ...   ...   
72824   NaN   NaN   NaN  NaN  NaN  0.40  0.26  0.34     0.52     0.48  2.76   
72825   NaN   NaN   NaN  NaN  NaN  0.41  0.27  0.33     0.52     0.48  2.76   
72826   NaN   NaN   NaN  NaN  NaN  0.52  0.25  0.22     0.56     0.44  2.94   
72827   NaN   NaN   NaN  NaN  NaN  0.35  0.28  0.37     0.52     0.48  2.76   
72828   NaN   NaN   NaN  NaN  NaN  0.17  0.23  0.60     0.58     0.42  3.03   

       HG_Exp  AG_Exp  Sup Exp  Derby  Res  
0        0.93    1.51    -0.58      0    1  
1        1.02    1.23    -0.21      0    2  
2        2.39    0.45     1.94      0    1  
3        1.58    0.95     0.63      0    1  
4        1.77    0.79     0.98      0    1  
...       ...     ...      ...    ...  ...  
72824    1.46    1.30     0.16      0    2  
72825    1.50    1.28     0.22      0    2  
72826    1.90    1.01     0.89      0    2  
72827    1.35    1.41    -0.06      0    2  
72828    0.86    2.17    -1.31      0    1  

[72829 rows x 37 columns]

#### Check whether there is any significant difference in the Goal Expectancy and Match Supremacy averages of the derby and non_derby DF's

In [15]:
print('Derbies Average Expected Sup: ',round(derbies['Sup Exp'].mean(),3))
print('Non-Derbies Average Expected Sup: ',round(non_derbies['Sup Exp'].mean(),3))
print('Derbies Average Goal Expectancy: ',round(derbies['Exp'].mean(),3))
print('Non-Derbies Average Goal Expectancy: ',round(non_derbies['Exp'].mean(),3))


Derbies Average Expected Sup:  0.339
Non-Derbies Average Expected Sup:  0.352
Derbies Average Goal Expectancy:  2.658
Non-Derbies Average Goal Expectancy:  2.646


Pre match average Supremacy's and Goal Expectancies are within around 0.01 goal, so no major differences  between the sub groups of our data sample 

# ___________________________________________________________________________________________________________________________________

## Data now prepared for descriptive statistical comparison

#### Observe general descriptive stats of the 'derby' DF

In [16]:
derbies.describe()

level_0          index         FTHG         FTAG         HTHG  \
count   3355.000000    3355.000000  3355.000000  3355.000000  3355.000000   
mean   34971.044709   61708.941282     1.444411     1.154993     0.664083   
std    25607.932484   50022.779113     1.216119     1.096887     0.815843   
min        1.000000    7797.000000     0.000000     0.000000     0.000000   
25%     9742.000000   17538.000000     1.000000     0.000000     0.000000   
50%    31839.000000   40258.000000     1.000000     1.000000     0.000000   
75%    62541.000000  119112.000000     2.000000     2.000000     1.000000   
max    76154.000000  145246.000000     8.000000     6.000000     4.000000   

              HTAG          HST          AST           HF           AF  \
count  3355.000000  3067.000000  3067.000000  3067.000000  3067.000000   
mean      0.490313     5.295729     4.369742    12.432344    12.788393   
std       0.699356     2.890747     2.469429     4.157323     4.140301   
min       0.000000     0.000000     0.000000     1.000000     1.000000   
25%       0.000000     3.000000     3.000000     9.000000    10.000000   
50%       0.000000     5.000000     4.000000    12.000000    13.000000   
75%       1.000000     7.000000     6.000000    15.000000    15.000000   
max       4.000000    20.000000    16.000000    33.000000    30.000000   

                HC           AC           HY           AY           HR  \
count  3067.000000  3067.000000  3067.000000  3067.000000  3067.000000   
mean      5.795892     4.797522     1.778937     2.090968     0.106293   
std       2.883033     2.603454     1.342530     1.394194     0.328743   
min       0.000000     0.000000     0.000000     0.000000     0.000000   
25%       4.000000     3.000000     1.000000     1.000000     0.000000   
50%       6.000000     5.000000     2.000000     2.000000     0.000000   
75%       8.000000     6.000000     3.000000     3.000000     0.000000   
max      19.000000    17.000000     7.000000     9.000000     3.000000   

                AR           TG          TST           TF           TC  \
count  3067.000000  3355.000000  3067.000000  3067.000000  3067.000000   
mean      0.124226     2.599404     9.665471    25.220737    10.593414   
std       0.355586     1.636968     3.992605     6.546177     3.533642   
min       0.000000     0.000000     0.000000     4.000000     2.000000   
25%       0.000000     1.000000     7.000000    21.000000     8.000000   
50%       0.000000     2.000000     9.000000    25.000000    10.000000   
75%       0.000000     4.000000    12.000000    29.000000    13.000000   
max       3.000000    10.000000    28.000000    56.000000    24.000000   

                TY           TR         HW_%          X_%         AW_%  \
count  3067.000000  3067.000000  2939.000000  2939.000000  2939.000000   
mean      3.869905     0.230518     0.426808     0.270660     0.302389   
std       2.151252     0.495256     0.119131     0.027278     0.107019   
min       0.000000     0.000000     0.080000     0.080000     0.030000   
25%       2.000000     0.000000     0.350000     0.260000     0.230000   
50%       4.000000     0.000000     0.420000     0.280000     0.300000   
75%       5.000000     0.000000     0.500000     0.290000     0.360000   
max      14.000000     5.000000     0.880000     0.340000     0.760000   

           Ov2.5_%      Un2.5_%          Exp       HG_Exp       AG_Exp  \
count  2939.000000  2939.000000  2936.000000  2936.000000  2936.000000   
mean      0.494515     0.505485     2.658154     1.498508     1.159387   
std       0.057477     0.057477     0.235732     0.370216     0.312978   
min       0.310000     0.220000     1.940000     0.520000     0.360000   
25%       0.460000     0.470000     2.530000     1.270000     0.960000   
50%       0.490000     0.510000     2.620000     1.450000     1.135000   
75%       0.530000     0.540000     2.790000     1.680000     1.310000   
max       0.780000     0.690000     3.820000     3.420000 

#### Observe general descriptive stats of the 'non-derbies' DF

In [17]:
non_derbies.describe()

level_0          index          FTHG          FTAG          HTHG  \
count  72829.000000   72829.000000  72829.000000  72829.000000  72826.000000   
mean   38235.249434   72646.764640      1.463675      1.159524      0.647406   
std    21801.033314   54893.446714      1.241167      1.103843      0.810183   
min        0.000000       0.000000      0.000000      0.000000      0.000000   
25%    19477.000000   19485.000000      1.000000      0.000000      0.000000   
50%    38282.000000   53763.000000      1.000000      1.000000      0.000000   
75%    56909.000000  135910.000000      2.000000      2.000000      1.000000   
max    76183.000000  145312.000000     10.000000      9.000000      7.000000   

               HTAG           HST           AST            HF            AF  \
count  72826.000000  66639.000000  66644.000000  66643.000000  66643.000000   
mean       0.506481      5.313195      4.299607     12.017166     12.514608   
std        0.717162      2.874686      2.514244      4.279834      4.308033   
min        0.000000      0.000000      0.000000      0.000000      0.000000   
25%        0.000000      3.000000      2.000000      9.000000     10.000000   
50%        0.000000      5.000000      4.000000     12.000000     12.000000   
75%        1.000000      7.000000      6.000000     15.000000     15.000000   
max        5.000000     27.000000     20.000000     48.000000     77.000000   

                 HC            AC            HY            AY            HR  \
count  66648.000000  66648.000000  66660.000000  66661.000000  66661.000000   
mean       5.860221      4.793992      1.510201      1.830831      0.080257   
std        2.946622      2.644652      1.246403      1.337785      0.286527   
min        0.000000      0.000000      0.000000      0.000000      0.000000   
25%        4.000000      3.000000      1.000000      1.000000      0.000000   
50%        6.000000      4.000000      1.000000      2.000000      0.000000   
75%        8.000000      6.000000      2.000000      3.000000      0.000000   
max       24.000000     21.000000     11.000000      9.000000      3.000000   

                 AR            TG           TST            TF            TC  \
count  66661.000000  72829.000000  66639.000000  66643.000000  66648.000000   
mean       0.112870      2.623200      9.612509     24.531774     10.654213   
std        0.341033      1.632101      3.886927      6.912619      3.564359   
min        0.000000      0.000000      0.000000      1.000000      0.000000   
25%        0.000000      1.000000      7.000000     20.000000      8.000000   
50%        0.000000      2.000000      9.000000     24.000000     10.000000   
75%        0.000000      4.000000     12.000000     29.000000     13.000000   
max        4.000000     12.000000     33.000000     90.000000     29.000000   

                 TY            TR          HW_%           X_%          AW_%  \
count  66660.000000  66661.000000  63327.000000  63327.000000  63327.000000   
mean       3.340984      0.193126      0.431747      0.267239      0.300962   
std        2.032963      0.461017      0.137085      0.034555      0.124040   
min        0.000000      0.000000      0.030000      0.050000      0.020000   
25%        2.000000      0.000000      0.350000      0.260000      0.220000   
50%        3.000000      0.000000      0.420000      0.280000      0.290000   
75%        5.000000      0.000000      0.510000      0.290000      0.360000   
max       15.000000      5.000000      0.920000      0.660000      0.880000   

            Ov2.5_%       Un2.5_%           Exp        HG_Exp        AG_Exp  \
count  63320.000000  63320.000000  63140.000000  63140.000000  63140.000000   
mean       0.492031      0.507970      2.645883      1.499108      1.146634   
std        0.057264      0.057262      0.229484      0.412276      0.362722   
min        0.280000      0.140000      1.840000      0.260000      0.250000   
25%        0.460000      0.480000      2.530000 

### Parameter comparisons in more detail and removing any unrecorded data within specific columns

#### 1. Total Match Goals:

In [18]:
# check if any rows have null values in the 'TG' column
df1['TG'].isna().sum()

0

In [19]:
derb_TG = round(derbies['TG'].mean(),3)
non_derb_TG = round(non_derbies['TG'].mean(),3)
all_matches_TG = round(df1['TG'].mean(),3)
derb_TG_pc = ((derb_TG - all_matches_TG) / all_matches_TG) *100
non_derb_TG_pc = ((non_derb_TG - all_matches_TG) / all_matches_TG) *100

print('Average Total Goals in Derbies is: ',derb_TG)
print('Average Total Goals in Non-Derbies is: ',non_derb_TG)
print('Average Total Goals in all matches combined is: ',all_matches_TG)
print("")
print(f'*** Derbies have {round(derb_TG_pc,2)} % more total goals than all matches combined')
print(f'*** Non-Derbies have {round(non_derb_TG_pc,2)} % more total goals than all matches combined')

Average Total Goals in Derbies is:  2.599
Average Total Goals in Non-Derbies is:  2.623
Average Total Goals in all matches combined is:  2.622

*** Derbies have -0.88 % more total goals than all matches combined
*** Non-Derbies have 0.04 % more total goals than all matches combined


#### 2. Total Home Goals:

In [20]:
derb_FTHG = round(derbies['FTHG'].mean(),3)
non_derb_FTHG = round(non_derbies['FTHG'].mean(),3)
all_matches_FTHG = round(df1['FTHG'].mean(),3)
derb_FTHG_pc = ((derb_FTHG - all_matches_FTHG) / all_matches_FTHG) *100
non_derb_FTHG_pc = ((non_derb_FTHG - all_matches_FTHG) / all_matches_FTHG) *100

print('Average Total HOME Goals in Derbies is: ',derb_FTHG)
print('Average Total HOME Goals in Non-Derbies is: ',non_derb_FTHG)
print('Average Total HOME Goals in all matches combined is: ',all_matches_FTHG)
print("")
print(f'*** Derbies have {round(derb_FTHG_pc,2)} % more HOME goals than all matches combined')
print(f'*** Non-Derbies have {round(non_derb_FTHG_pc,2)} % more HOME goals than all matches combined')

Average Total HOME Goals in Derbies is:  1.444
Average Total HOME Goals in Non-Derbies is:  1.464
Average Total HOME Goals in all matches combined is:  1.463

*** Derbies have -1.3 % more HOME goals than all matches combined
*** Non-Derbies have 0.07 % more HOME goals than all matches combined


#### 3. Total Away Goals:

In [21]:
derb_FTAG = round(derbies['FTAG'].mean(),3)
non_derb_FTAG = round(non_derbies['FTAG'].mean(),3)
all_matches_FTAG = round(df1['FTAG'].mean(),3)
derb_FTAG_pc = ((derb_FTAG - all_matches_FTAG) / all_matches_FTAG) *100
non_derb_FTAG_pc = ((non_derb_FTAG - all_matches_FTAG) / all_matches_FTAG) *100

print('Average Total AWAY Goals in Derbies is: ',derb_FTAG)
print('Average Total AWAY Goals in Non-Derbies is: ',non_derb_FTAG)
print('Average Total AWAY Goals in all matches combined is: ',all_matches_FTAG)
print("")
print(f'*** Derbies have {round(derb_FTAG_pc,2)} % more AWAY goals than all matches combined')
print(f'*** Non-Derbies have {round(non_derb_FTAG_pc,2)} % more AWAY goals than all matches combined')

Average Total AWAY Goals in Derbies is:  1.155
Average Total AWAY Goals in Non-Derbies is:  1.16
Average Total AWAY Goals in all matches combined is:  1.159

*** Derbies have -0.35 % more AWAY goals than all matches combined
*** Non-Derbies have 0.09 % more AWAY goals than all matches combined


#### 4. Home Wins

In [22]:
h_win_pc_derb = round((len(derbies[derbies['Res'] == 1]) / len(derbies)) * 100,3)
h_win_pc_n_derb = round((len(non_derbies[non_derbies['Res'] == 1]) / len(non_derbies)) * 100,3)
h_win_pc_all = round((len(df1[df1['Res'] == 1]) / len(df1)) * 100,3)
h_win_pc_diff = (h_win_pc_derb - h_win_pc_all) 
h_win_pc_n_derb_diff = (h_win_pc_n_derb - h_win_pc_all) 

print('Average Total Home Wins in Derbies is: ',h_win_pc_derb)
print('Average Total Home Wins in Non-Derbies is: ',h_win_pc_n_derb)
print('Average Total Home Wins in all matches combined is: ',h_win_pc_all)
print("")
print(f'*** Derbies have {round(h_win_pc_diff,2)} % more Home Wins than all matches combined')
print(f'*** Non-Derbies have {round(h_win_pc_n_derb_diff,2)} % more Home Wins than all matches combined')

Average Total Home Wins in Derbies is:  42.623
Average Total Home Wins in Non-Derbies is:  43.759
Average Total Home Wins in all matches combined is:  43.709

*** Derbies have -1.09 % more Home Wins than all matches combined
*** Non-Derbies have 0.05 % more Home Wins than all matches combined


#### 5. Away Wins

In [23]:
a_win_pc_derb = round((len(derbies[derbies['Res'] == 2]) / len(derbies)) * 100, 3)
a_win_pc_n_derb = round((len(non_derbies[non_derbies['Res'] == 2]) / len(non_derbies)) * 100, 3)
a_win_pc_all = round((len(df1[df1['Res'] == 2]) / len(df1)) * 100, 3)
a_win_pc_derb_diff = (a_win_pc_derb - a_win_pc_all) 
a_win_pc_n_derb_diff = (a_win_pc_n_derb - a_win_pc_all) 

print('Average Total Away Win % in Derbies is: ',a_win_pc_derb)
print('Average Total Away Win % in Non-Derbies is: ',a_win_pc_n_derb)
print('Average Total Away Win % in all matches combined is: ',a_win_pc_all)
print("")
print(f'*** Derbies have {round(a_win_pc_derb_diff,2)} % more Away Wins than all matches combined')
print(f'*** Non-Derbies have {round(a_win_pc_n_derb_diff,2)} % more Away Wins than all matches combined')

Average Total Away Win % in Derbies is:  29.359
Average Total Away Win % in Non-Derbies is:  29.875
Average Total Away Win % in all matches combined is:  29.853

*** Derbies have -0.49 % more Away Wins than all matches combined
*** Non-Derbies have 0.02 % more Away Wins than all matches combined


#### 6. Draws

In [24]:
x_pc_derb = round((len(derbies[derbies['Res'] == 0]) / len(derbies)) * 100, 3)
x_pc_n_derb = round((len(non_derbies[non_derbies['Res'] == 0]) / len(non_derbies)) * 100, 3)
x_pc_all = round((len(df1[df1['Res'] == 0]) / len(df1)) * 100, 3)
x_pc_derb_diff = (x_pc_derb - x_pc_all) 
x_pc_n_derb_diff = (x_pc_n_derb - x_pc_all) 

print('Average Draw % in Derbies is: ',x_pc_derb)
print('Average Draw % in Non-Derbies is: ',x_pc_n_derb)
print('Average Draw % in all matches combined is: ',x_pc_all)
print("")
print(f'*** Derbies have {round(x_pc_derb_diff,2)} % more Draws than all matches combined')
print(f'*** Non-Derbies have {round(x_pc_n_derb_diff,2)} % more Draws than all matches combined')

Average Draw % in Derbies is:  28.018
Average Draw % in Non-Derbies is:  26.366
Average Draw % in all matches combined is:  26.439

*** Derbies have 1.58 % more Draws than all matches combined
*** Non-Derbies have -0.07 % more Draws than all matches combined


#### 7. Total Shots on Target

In [25]:
# check if any rows have null values in the 'TST' column
print('No of rows missing shots data: ',df1['TST'].isna().sum())

No of rows missing shots data:  6478


In [26]:
# drop all rows with null values in the 'TST' columns
df1 = df1.dropna(subset=['TST'])
derbies = derbies.dropna(subset=['TST'])
non_derbies = non_derbies.dropna(subset=['TST'])

# check they've been dropped
print('No of rows missing shots data: ',df1['TST'].isna().sum())

No of rows missing shots data:  0


In [27]:
derb_TST = round(derbies['TST'].mean(),3)
non_derb_TST = round(non_derbies['TST'].mean(),3)
all_matches_TST = round(df1['TST'].mean(),3)
derb_SOT_pc = ((derb_TST - all_matches_TST) / all_matches_TST) *100
non_derb_SOT_pc = ((non_derb_TST - all_matches_TST) / all_matches_TST) *100

print('Average Total SOT in Derbies is: ',derb_TST)
print('Average Total SOT in Non-Derbies is: ',non_derb_TST)
print('Average Total SOT in all matches combined is: ',all_matches_TST)
print("")
print(f'*** Derbies have {round(derb_SOT_pc,2)} % more SOT than all matches combined')
print(f'*** Non-Derbies have {round(non_derb_SOT_pc,2)} % more SOT than all matches combined')

Average Total SOT in Derbies is:  9.665
Average Total SOT in Non-Derbies is:  9.613
Average Total SOT in all matches combined is:  9.615

*** Derbies have 0.52 % more SOT than all matches combined
*** Non-Derbies have -0.02 % more SOT than all matches combined


#### 8. Total Corners:

In [28]:
# check if any rows have null values in the 'TST' column
df1['TC'].isna().sum()

0

In [29]:
# drop all rows with null values in the 'TC' columns
df1 = df1.dropna(subset=['TC'])
derbies = derbies.dropna(subset=['TC'])
non_derbies = non_derbies.dropna(subset=['TC'])
# check they've been dropped
derbies['TC'].isna().sum()

0

In [30]:
derb_TC = round(derbies['TC'].mean(),3)
non_derb_TC = round(non_derbies['TC'].mean(),3)
all_matches_TC = round(df1['TC'].mean(),3)
derb_TC_pc = ((derb_TC - all_matches_TC) / all_matches_TC) *100
non_derb_TC_pc = ((non_derb_TC - all_matches_TC) / all_matches_TC) *100

print('Average Total Corners in Derbies is: ',derb_TC)
print('Average Total Corners in Non-Derbies is: ',non_derb_TC)
print('Average Total Corners in all matches combined is: ',all_matches_TC)
print("")
print(f'*** Derbies have {round(derb_TC_pc,2)} % more Corners than all matches combined')
print(f'*** Non-Derbies have {round(non_derb_TC_pc,2)} % more Corners than all matches combined')

Average Total Corners in Derbies is:  10.593
Average Total Corners in Non-Derbies is:  10.654
Average Total Corners in all matches combined is:  10.651

*** Derbies have -0.54 % more Corners than all matches combined
*** Non-Derbies have 0.03 % more Corners than all matches combined


#### 9. Total Fouls:

In [31]:
# check if any rows have null values in the 'TF' column
print('No of rows missing fouls data: ',df1['TF'].isna().sum())

No of rows missing fouls data:  5


In [32]:
# drop all rows with null values in the 'TF' columns
df1 = df1.dropna(subset=['TF'])
derbies = derbies.dropna(subset=['TF'])
non_derbies = non_derbies.dropna(subset=['TF'])

# check they've been dropped
print('No of rows missing fouls data: ',df1['TF'].isna().sum())

No of rows missing fouls data:  0


In [33]:
derb_TF = round(derbies['TF'].mean(),3)
non_derb_TF = round(non_derbies['TF'].mean(),3)
all_matches_TF = round(df1['TF'].mean(),3)
derb_TF_pc = ((derb_TF - all_matches_TF) / all_matches_TF) *100
non_derb_TF_pc = ((non_derb_TF - all_matches_TF) / all_matches_TF) *100

print('Average Total Fouls in Derbies is: ',derb_TF)
print('Average Total Fouls in Non-Derbies is: ',non_derb_TF)
print('Average Total Fouls in all matches combined is: ',all_matches_TF)
print("")
print(f'*** Derbies have {round(derb_TF_pc,2)} % more Fouls than all matches combined')
print(f'*** Non-Derbies have {round(non_derb_TF_pc,2)} % more Fouls than all matches combined')

Average Total Fouls in Derbies is:  25.221
Average Total Fouls in Non-Derbies is:  24.53
Average Total Fouls in all matches combined is:  24.56

*** Derbies have 2.69 % more Fouls than all matches combined
*** Non-Derbies have -0.12 % more Fouls than all matches combined


#### 9. Total Yellows:

In [34]:
# check if any rows have null values in the 'TY' column
print('No of rows missing yellows data: ',df1['TY'].isna().sum())

No of rows missing yellows data:  1


In [35]:
# drop all rows with null values in the 'TY' columns
df1 = df1.dropna(subset=['TY'])
derbies = derbies.dropna(subset=['TY'])
non_derbies = non_derbies.dropna(subset=['TY'])

# check they've been dropped
print('No of rows missing yellows data: ',df1['TY'].isna().sum())

No of rows missing yellows data:  0


In [36]:
derb_TY = round(derbies['TY'].mean(),3)
non_derb_TY = round(non_derbies['TY'].mean(),3)
all_matches_TY = round(df1['TY'].mean(),3)
derb_TY_pc = ((derb_TY - all_matches_TY) / all_matches_TY) *100
non_derb_TY_pc = ((non_derb_TY - all_matches_TY) / all_matches_TY) *100

print('Average Total Yellows in Derbies is: ',derb_TY)
print('Average Total Yellows in Non-Derbies is: ',non_derb_TY)
print('Average Total Yellows in all matches combined is: ',all_matches_TY)
print("")
print(f'*** Derbies have {round(derb_TY_pc,2)} % more Yellows than all matches combined')
print(f'*** Non-Derbies have {round(non_derb_TY_pc,2)} % more Yellows than all matches combined')

Average Total Yellows in Derbies is:  3.87
Average Total Yellows in Non-Derbies is:  3.341
Average Total Yellows in all matches combined is:  3.364

*** Derbies have 15.04 % more Yellows than all matches combined
*** Non-Derbies have -0.68 % more Yellows than all matches combined


#### 10. Total Reds:

In [37]:
# check if any rows have null values in the 'TR' column
print('No of rows missing reds data: ',df1['TR'].isna().sum())

No of rows missing reds data:  0


In [38]:
derb_TR = round(derbies['TR'].mean(),3)
non_derb_TR = round(non_derbies['TR'].mean(),3)
all_matches_TR = round(df1['TR'].mean(),3)
derb_TR_pc = ((derb_TR - all_matches_TR) / all_matches_TR) *100
non_derb_TR_pc = ((non_derb_TR - all_matches_TR) / all_matches_TR) *100

print('Average Total Reds in Derbies is: ',derb_TR)
print('Average Total Reds in Non-Derbies is: ',non_derb_TR)
print('Average Total Reds in all matches combined is: ',all_matches_TR)
print("")
print(f'*** Derbies have {round(derb_TR_pc,2)} % more Reds than all matches combined')
print(f'*** Non-Derbies have {round(non_derb_TR_pc,2)} % more Reds than all matches combined')

Average Total Reds in Derbies is:  0.231
Average Total Reds in Non-Derbies is:  0.193
Average Total Reds in all matches combined is:  0.195

*** Derbies have 18.46 % more Reds than all matches combined
*** Non-Derbies have -1.03 % more Reds than all matches combined


## Conclusions:

#### This analysis has identified some significant and some minor differences between the outcome of various match parameters between derby and non-derby matches
#### The findings can be summarised as:
- 0.92% fewer total goals
- 1.37% fewer home goals
- 0.44% fewer away goals
- 1.14% fewer home wins
- 0.51% fewer away wins
- 0.57% fewer corners
- 1.65% more draws
- 0.54% more shots on target
- 2.81% more fouls
- 15.72% more yellow cards
- 19.49% more red cards

These differences are based on an average magnitude derby. Rivalries of greater intensity will have higher outcome anomolies and vice versa. Further research can be carried out on a list of identified higher order rivalries albeit at the expense of a smaller dataset from which to draw conclusions.